### DISCUSSION TITLE
design in `PRBProxy.sol#execute`

### DISCUSSION THREAD

**Discussion by wiasliaw** [PaulRBerg/prb-proxy#194] [Q&A] [answered]
In other similar kind of proxy implementation, like diamond, it only makes delegatecall to the implementation which had registered. It is weird that address which has permission can make delegatecall to any address to execute arbitrary code.

**Comment by PaulRBerg** [comment, marked as answer]
This project should have been named [PRBForwarder](https://github.com/PaulRBerg/prb-proxy/discussions/178) instead of PRBProxy.

> It is weird that address which has permission can make delegatecall to any address to execute arbitrary code.

That's the main reason why I created this. See the README.
